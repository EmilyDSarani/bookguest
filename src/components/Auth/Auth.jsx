// import { Link, useLocation } from 'react-router-dom'
// import { useUser } from '../../context/UserContext'

import { useLocation } from "react-router"

export default function Auth() {
    const { user, setUser } = useUser()
    const location = useLocation()

    const { from } = location.state || {from: { pathname: '/'} };




    return (
        <>
        <fieldset>
        <legend>Sign In</legend>
        <form>
          <label htmlFor="username" >
            Username
          </label>
          <input
            id="username"
            type="text"
            name="username"
            required
          />
          <label htmlFor="password" >
            Password
          </label>
          <input
            id="password"
            type="password"
            name="password"
            required
          />
          <button
            type="submit"
          >
            Sign In
          </button>
        </form>
        <p>Error message goes here</p>
      </fieldset>
      </>
    )
}
