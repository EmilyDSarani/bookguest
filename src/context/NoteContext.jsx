import { useContext, createContext, useState } from 'react'

//Create Context
//Remember that this is the start of our recipe
const NoteContext = createContext()

//Create Provider
//Remember that the provider is giving us access to the recipe book
const NoteProvider = ({ children }) =>{
    const [notes, setNotes] = useState(
        //might need to drop a useEffect here later on


    {})
    return <NoteContext.Provider value={{notes, setNotes}}>{children}</NoteContext.Provider>   
}

//Create Custom Hook
const useNote = () => {
    const context = useContext(NoteContext)

    if(context === undefined){
        throw new Error('useUser needs to be defined in the NoteContext Provider')
    }

    return context
}

export {NoteProvider, useNote}