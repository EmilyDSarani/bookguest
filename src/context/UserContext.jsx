import { useContext, createContext, useState, useEffect } from 'react'

//Create Context
//Remember that this is the start of our recipe
const UserContext = createContext()

//Create Provider
//Remember that the provider is giving us access to the recipe book
const UserProvider = ({ children }) =>{
    const [user, setUser] = useState(
        //might need to drop a useEffect here later on


    {})
    return <UserContext.Provider value={{user, setUser}}>{children}</UserContext.Provider>   
}

//Create Custom Hook
const useUser = () => {
    const context = useContext(UserContext)

    if(context === undefined){
        throw new Error('useUser needs to be defined in the UserContext Provider')
    }

    return context
}

export {UserProvider, useUser}