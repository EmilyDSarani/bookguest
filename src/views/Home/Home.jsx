import React from 'react'
import NoteList from '../../components/NoteList/NoteList'
import GuestBook from '../../components/GuestBook/GuestBook'
import { NoteProvider } from '../../context/NoteContext'

// we wrap <GuessBook /> and <NoteList /> in the NoteProvider tag that was imported in so that everything that is in App will get access to that specific state 
export default function Home() {
    return (
        <div>
            <NoteProvider>
            <GuestBook />
            <NoteList />
            </NoteProvider>
        </div>
    )
}
