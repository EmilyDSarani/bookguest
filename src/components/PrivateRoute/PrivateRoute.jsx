import { Redirect, Route } from 'react-router-dom'
import { useUser } from '../../context/UserContext'

export default function PrivateRoute({children, ...routeProps}) {
    const { user } = useUser()

//spreading gives us access to magic props that come from react

    return (
        <>
           <Route
           {...routeProps}
           render={({ location }) => user ? children : <Redirect to={{ pathname: '/login', state: { from: location}} }
           /> 
        }
        />    
        </>
    )
}
