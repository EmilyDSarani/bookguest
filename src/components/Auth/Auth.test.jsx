import { render } from '@testing-library/react';
import Auth from './Auth'
import Home from '../../views/Home/Home'
import PrivateRoute from '../PrivateRoute/PrivateRoute';
import { UserProvider } from '../../context/UserContext';
import { MemoryRouter, Route, Switch } from 'react-router';




it('should pull up the Guestbook', async () =>{
    const { container } = render(
        <UserProvider>
            <MemoryRouter initialEntries={["/login"]}>
                <Switch>
                <Route exact path ="/login">
                    <Auth />
                </Route>
                <PrivateRoute exact path= "/">
                    <Home />
                </PrivateRoute>
                </Switch>
            </MemoryRouter>
        </UserProvider>
    )
    expect(container).toMatchSnapshot();
    }
)
 //work please