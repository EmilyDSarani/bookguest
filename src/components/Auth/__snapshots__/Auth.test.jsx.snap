// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should pull up the Guestbook 1`] = `
<div>
  <fieldset>
    <legend>
      Sign In
    </legend>
    <form>
      <label
        for="email"
      >
        Email
      </label>
      <input
        id="email"
        name="email"
        type="text"
        value=""
      />
       
      <label
        for="password"
      >
        Password
      </label>
      <input
        id="password"
        name="password"
        type="password"
        value=""
      />
      <button
        type="submit"
      >
        Sign In
      </button>
    </form>
  </fieldset>
</div>
`;
